### PR TITLE
Add another example of guards indentation

### LINKED ISSUE #41: How to indent multiline guard clause?
Context: https://github.com/uohzxela/ex_format/pull/63

Style 1:

```
  defp parenless_capture?({atom, _, _})
       when is_atom(atom) and
       atom not in unquote(@unary_ops) and
       atom not in unquote(@binary_ops) do
    true
  end
```

Style 2:

```
  defp parenless_capture?({atom, _, _})
       when is_atom(atom) and
            atom not in unquote(@unary_ops) and
            atom not in unquote(@binary_ops) do
    true
  end
```

Which one is preferred?

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes #41.

### CODE REVIEW

**Review by ericmj — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -172,6 +172,13 @@
     # ...
   end
 
+  defp parenless_capture?({atom, _meta, _args})
```
Comment (lexmag) [inline README.md]: Maybe `atom` –> `op`?

**Review by lexmag — COMMENTED** [review]
I think we should describe `when` indentation in more details, current

> The only exceptions are `when` in guards and `|>`, which go on the beginning of the line and should be indented at the same level as their left-hand side.

wording sounds like `when` should be indented as the following:

```elixir
defmacro dngettext(domain, msgid, msgid_plural, count)
when is_binary(msgid) and is_binary(msgid_plural) do
```

_Perhaps it is from old days when such indentation indeed was preferable._

**Comment by michalmuskala** [comment]
@lexmag I think the comment about indenting `when` refers to the situation where you have multiple `whens`:

```elixir
defmacro dngettext(domain, msgid, msgid_plural, count)
         when is_binary(msgid) and is_binary(msgid_plural)
         when is_list(msgid) and is_list(msgid_plural) do
```

If we always indent, it would need to look like this:
```elixir
defmacro dngettext(domain, msgid, msgid_plural, count)
         when is_binary(msgid) and is_binary(msgid_plural)
           when is_list(msgid) and is_list(msgid_plural) do
```

**Comment by lexmag** [comment]
@michalmuskala makes sense. 👍
Then I withdraw my comment on that.

**Comment by michalmuskala** [comment]
Maybe we could rephrase it or add an example to make it clearer.

**Comment by whatyouhide** [comment]
I added an example that I think shows what we mean without adding too many rules and stuff. Let me know what you think.

**Review by lexmag — APPROVED** [review]

**Pull request merged**